crates/libs/bindgen/src/gen.rs

```diff
@@ -839,67 +839,51 @@ impl<'a> Gen<'a> {
 
         for (position, param) in params.iter().enumerate() {
             match kind {
-                SignatureKind::Query((object, guid)) => {
-                    if object == position {
-                        tokens.combine(&quote! { &mut result__ as *mut _ as *mut _, });
-                        continue;
+                SignatureKind::Query(query) if query.object == position => {
+                    tokens.combine(&quote! { &mut result__ as *mut _ as *mut _, });
+                }
+                SignatureKind::QueryOptional(query) if query.object == position => {
+                    tokens.combine(&quote! { result__ as *mut _ as *mut _, });
+                }
+                SignatureKind::Query(query) | SignatureKind::QueryOptional(query) if query.guid == position => {
+                    tokens.combine(&quote! { &<T as ::windows::core::Interface>::IID, });
+                }
+                _ => {
+                    let name = self.param_name(param.def);
+                    if let ArrayInfo::Fixed(fixed) = param.array_info {
```

Comment:
Instead of the chained `if let ArrayInfo` and `continue` calls you can turn this all into a match expression.

Reply:
There seem to be quite a few such candidates. I can tackle these separately.